src/EnergyPlus/PVWatts.hh

```diff
@@ -0,0 +1,430 @@
+//     written permission.
+//
+// (4) Use of EnergyPlus(TM) Name. If Licensee (i) distributes the software in stand-alone form
+//     without changes from the version obtained under this License, or (ii) Licensee makes a
+//     reference solely to the software portion of its product, Licensee must refer to the
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef PVWatts_hh_INCLUDED
+#define PVWatts_hh_INCLUDED
+
+// C++ Headers
+#include <string>
+#include <map>
+#include <memory>
+
+// ObjexxFCL Headers
+
+// EnergyPlus Headers
+#include <EnergyPlus.hh>
+#include <DataSurfaces.hh>
+#include <InputProcessor.hh>
+
+namespace EnergyPlus {
+
+namespace PVWatts {
+
+	const Real64 AOI_MIN(0.5);
+	const Real64 AOI_MAX(89.5);
+	const Real64 DTOR(DataGlobals::DegToRadians);
+	enum RADMODE {DN_DF, DN_GH, GH_DF, POA_R, POA_P};
+	const Real64 SMALL(1e-6);
+
+	enum class ModuleType {
+		STANDARD,
+		PREMIUM,
+		THIN_FILM,
+	};
+
+	enum class ArrayType {
+		FIXED_OPEN_RACK,
+		FIXED_ROOF_MOUNTED,
+		ONE_AXIS,
+		ONE_AXIS_BACKTRACKING,
+		TWO_AXIS,
+	};
+
+	enum class GeometryType {
+		TILT_AZIMUTH,
+		SURFACE,
+	};
+
+	struct DCPowerOutput {
+		Real64 poa; // Plane of array irradiance
+		Real64 tpoa; // Transmitted plane of array irradiance
+		Real64 pvt; // PV Cell temperature
+		Real64 dc; // DC power output
+	};
+
+	struct IrradianceOutput {
+		Real64 solazi;
+		Real64 solzen;
+		Real64 solalt;
+		Real64 aoi;
+		Real64 stilt;
+		Real64 sazi;
+		Real64 rot;
+		Real64 btd;
+		Real64 ibeam;
+		Real64 iskydiff;
+		Real64 ignddiff;
+		int sunup;
+	};
+
+	// From here down is from the SAM Simulation core library
```

Comment:
So...if the SAM simulation core library has an update, how does that get patched here?  Is the SAM core library available as something we could've instead dropped into the `third_party/` folder so updates could be dropped right in? (Assuming their API is stable, of course.)